pkg/apis/db/v1alpha1/schema.go

```diff
@@ -0,0 +1,41 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// DatabaseSchema is a specification for a DatabaseSchema resource
+type DatabaseSchema struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   DatabaseSchemaSpec   `json:"spec"`
+	Status DatabaseSchemaStatus `json:"status"`
+}
+
+// DatabaseSchemaSpec is the spec for a DatabaseSchema resource
+type DatabaseSchemaSpec struct {
+	Database string `json:"database"`
```

Comment:
This should support a `database` that points to a `Database` object, but it should also be able to be used independently by specifying just a `dsn` or `dsnFrom` and a `serverType`. See how `Database` does it.